### PR TITLE
[SYCL][Graph] device_query.cpp: add test for in order queue

### PR DESCRIPTION
**Pull request by guoyejun**

### CODE REVIEW

**Review by EwanC — CHANGES_REQUESTED** [review]
I don't understand this change, SYCL-Graph support isn't dependent on the in-orderness of a queue.

File: sycl/test-e2e/Graph/device_query.cpp [hunk 1]
```diff
@@ -31,3 +29,11 @@ int main() {
     assert(SupportsGraphs == exp_ext::graph_support_level::unsupported);
   }
 }
+
+int main() {
+  queue Queue;
+  test(Queue);
+
+  queue ioq{sycl::gpu_selector_v, {sycl::property::queue::in_order()}};
```
Comment (EwanC) [inline sycl/test-e2e/Graph/device_query.cpp:37]: `sycl::gpu_selector_v` seems like it will prohibit CPU devices, for example OpenCL CPU which can be used

**Comment by guoyejun** [comment]
> I don't understand this change, SYCL-Graph support isn't dependent on the in-orderness of a queue, just the device -
> 
> https://github.com/intel/llvm/blob/58ade846be103aa5ec3f1b944b619f3620f55bb3/sycl/test-e2e/Graph/graph_common.hpp#L462
> 
> 
> and all we do in the new `test` function is also query the queue device.

from test perspective, the origin code just tests with the out of order queue, we need to also test graph feature with in order queue which is used in some domains.

File: sycl/test-e2e/Graph/device_query.cpp (hunk 1 above)
Comment (guoyejun) [reply sycl/test-e2e/Graph/device_query.cpp]: sycl::gpu_selector_v is removed

**Comment by guoyejun** [comment]
looks that it is not meaningful to extend the test coverage to in order queue in this test, I'll see other tests.

**Comment by EwanC** [comment]
> looks that it is not meaningful to extend the test coverage to in order queue in this test, I'll see other tests.

We've got quite a lot of tests that test recording a graph with the in_order property in https://github.com/intel/llvm/blob/sycl/sycl/unittests/Extensions/CommandGraph.cpp and also in test-e2e

```bash
dpcpp/sycl/test-e2e/Graph$ git grep in_order
RecordReplay/dotp_in_order.cpp:  property_list Properties{property::queue::in_order{}};
RecordReplay/dotp_in_order_pause.cpp:  property_list Properties{property::queue::in_order{}};
RecordReplay/dotp_in_order_with_empty_nodes.cpp:  property_list Properties{property::queue::in_order{}};
RecordReplay/dotp_multiple_queues.cpp:  property_list Properties{property::queue::in_order{}};
RecordReplay/host_task_in_order.cpp:  queue Queue{property::queue::in_order{}};
RecordReplay/in_order_queue_with_host_managed_dependencies.cpp:  queue Queue{sycl::property::queue::in_order{}};
RecordReplay/in_order_queue_with_host_managed_dependencies_memcpy.cpp:  queue Queue{sycl::property::queue::in_order{}};
RecordReplay/in_order_queue_with_host_managed_dependencies_memset.cpp:  queue Queue{sycl::property::queue::in_order{}};
RecordReplay/sub_graph_in_order.cpp:  property_list Properties{property::queue::in_order{}};
RecordReplay/usm_copy_in_order.cpp:  property_list Properties{property::queue::in_order{}};
```

But can't see any that test creating an out-of-order graph that is then executed on an in-order queue, so those would be useful.

**Comment by guoyejun** [comment]
> > looks that it is not meaningful to extend the test coverage to in order queue in this test, I'll see other tests.
> 
> We've got quite a lot of tests that test recording a graph with the in_order property in https://github.com/intel/llvm/blob/sycl/sycl/unittests/Extensions/CommandGraph.cpp and also in test-e2e
> 
> ```shell
> dpcpp/sycl/test-e2e/Graph$ git grep in_order
> RecordReplay/dotp_in_order.cpp:  property_list Properties{property::queue::in_order{}};
> RecordReplay/dotp_in_order_pause.cpp:  property_list Properties{property::queue::in_order{}};
> RecordReplay/dotp_in_order_with_empty_nodes.cpp:  property_list Properties{property::queue::in_order{}};
> RecordReplay/dotp_multiple_queues.cpp:  property_list Properties{property::queue::in_order{}};
> RecordReplay/host_task_in_order.cpp:  queue Queue{property::queue::in_order{}};
> RecordReplay/in_order_queue_with_host_managed_dependencies.cpp:  queue Queue{sycl::property::queue::in_order{}};
> RecordReplay/in_order_queue_with_host_managed_dependencies_memcpy.cpp:  queue Queue{sycl::property::queue::in_order{}};
> RecordReplay/in_order_queue_with_host_managed_dependencies_memset.cpp:  queue Queue{sycl::property::queue::in_order{}};
> RecordReplay/sub_graph_in_order.cpp:  property_list Properties{property::queue::in_order{}};
> RecordReplay/usm_copy_in_order.cpp:  property_list Properties{property::queue::in_order{}};
> ```
> 
> But can't see any that test creating an out-of-order graph that is then executed on an in-order queue, so those would be useful.

thanks for the info.

we are always using in-order queue (maybe different in-order queues) for warmup, record and replay. Recording in out-of-order queue and then replaying in in-order queue is currently not our focus.

closing this PR.

**Pull request closed without merging**